pkg/cloudprovider/providers/azure/azure_loadbalancer.go

```diff
@@ -90,32 +95,42 @@ func (az *Cloud) GetLoadBalancer(clusterName string, service *v1.Service) (statu
 	}, true, nil
 }
 
-func (az *Cloud) determinePublicIPName(clusterName string, service *v1.Service) (string, error) {
+func (az *Cloud) determinePublicIPName(clusterName string, service *v1.Service) (pipName string, pipResourceGroup string, err error) {
 	loadBalancerIP := service.Spec.LoadBalancerIP
 	if len(loadBalancerIP) == 0 {
-		return getPublicIPName(clusterName, service), nil
+		return getPublicIPName(clusterName, service), "", nil
+	}
+
+	// Override the cluster resource group if public IP resource ID presents
+	resourceGroup := az.ResourceGroup
+	pipID := publicIPAddressResourceID(service)
+	if pipID != "" {
+		resourceGroup, err = parseResourceGroupNameFromID(pipID)
+		if err != nil {
+			return "", "", err
+		}
 	}
 
 	az.operationPollRateLimiter.Accept()
-	glog.V(10).Infof("PublicIPAddressesClient.List(%v): start", az.ResourceGroup)
-	list, err := az.PublicIPAddressesClient.List(az.ResourceGroup)
-	glog.V(10).Infof("PublicIPAddressesClient.List(%v): end", az.ResourceGroup)
+	glog.V(10).Infof("PublicIPAddressesClient.List(%v): start", resourceGroup)
+	list, err := az.PublicIPAddressesClient.List(resourceGroup)
+	glog.V(10).Infof("PublicIPAddressesClient.List(%v): end", resourceGroup)
 	if err != nil {
-		return "", err
+		return "", "", err
 	}
 
 	if list.Value != nil {
 		for ix := range *list.Value {
 			ip := &(*list.Value)[ix]
 			if ip.PublicIPAddressPropertiesFormat.IPAddress != nil &&
 				*ip.PublicIPAddressPropertiesFormat.IPAddress == loadBalancerIP {
-				return *ip.Name, nil
+				return *ip.Name, resourceGroup, nil
 			}
 		}
 	}
 	// TODO: follow next link here? Will there really ever be that many public IPs?
 
-	return "", fmt.Errorf("user supplied IP Address %s was not found", loadBalancerIP)
+	return "", resourceGroup, fmt.Errorf("user supplied IP Address %s was not found", loadBalancerIP)
```

Comment:
I'm not sure of the Go idiom here (I'm pretty new to this too) but I don't think consumers are going to use the resource group in an error return, so would it be more idiomatic to return `""` for the resource group here?